test/DynamicDependency/Test_Win32/Test_GetCurrentPackageInfo.cpp

```diff
@@ -0,0 +1,374 @@
+            Assert::AreEqual(expectedRC, rc);
+            Assert::IsTrue(minExpectedBufferSize <= bufferSize);
+            Assert::AreEqual(expectedCount, count);
+        }
+
+        void VerifyGetCurrentPackageInfo3(
+            const UINT32 flags,
+            const HRESULT expectedHR = HRESULT_FROM_WIN32(APPMODEL_ERROR_NO_PACKAGE),
+            const UINT32 minExpectedBufferSize = 0,
+            const UINT32 expectedCount = 0)
+        {
+            if (!IsGetCurrentPackageInfo3Supported())
+            {
+                return;
+            }
+
+            const auto packageInfoType{ PackageInfoType_PackageInfoInstallPath };
+            UINT32 bufferSize{};
+            UINT32 count{};
+            const auto hr{ m_getCurrentPackageInfo3(flags, packageInfoType, &bufferSize, nullptr, &count) };
+            PrintGetCurrentPackageInfoN(L"GetCurrentPackageInfo3", flags, hr, bufferSize, count, expectedHR, minExpectedBufferSize, expectedCount);
+            Assert::AreEqual(expectedHR, hr);
+            Assert::IsTrue(minExpectedBufferSize <= bufferSize);
+            Assert::AreEqual(expectedCount, count);
+        }
+
+        void VerifyGetCurrentPackageInfo123(
+            const UINT32 flags,
+            const HRESULT expectedHR12 = HRESULT_FROM_WIN32(APPMODEL_ERROR_NO_PACKAGE),
+            const UINT32 minExpectedBufferSize12 = 0,
+            const UINT32 expectedCount12 = 0,
+            const HRESULT expectedHR3 = HRESULT_FROM_WIN32(APPMODEL_ERROR_NO_PACKAGE),
+            const UINT32 minExpectedBufferSize3 = 0,
+            const UINT32 expectedCount3 = 0)
+        {
+            VerifyGetCurrentPackageInfo1(flags, expectedHR12, minExpectedBufferSize12, expectedCount12);
+            VerifyGetCurrentPackageInfo2(flags, expectedHR12, minExpectedBufferSize12, expectedCount12);
+            VerifyGetCurrentPackageInfo3(flags, expectedHR3, minExpectedBufferSize3, expectedCount3);
+        }
+
+        void VerifyGenerationId(
+            const UINT32 expectedGenerationId,
+            const HRESULT expectedHR = HRESULT_FROM_WIN32(APPMODEL_ERROR_NO_PACKAGE))
+        {
+            if (!IsGetCurrentPackageInfo3Supported())
+            {
+                return;
+            }
+
+            UINT32 generationId{};
+            UINT32 bufferSize{ static_cast<UINT32>(sizeof(generationId)) };
+            const auto hr{ m_getCurrentPackageInfo3(0, PackageInfoType_PackageInfoGeneration, &bufferSize, &generationId, nullptr) };
+            auto message{ wil::str_printf<wil::unique_process_heap_string>(L"Get...GenerationId: hr:0x%X id:%u\n"
+                                                                           L"          Expected: hr:0x%X id:%u",
+                                                                           hr, generationId, expectedHR, expectedGenerationId) };
+            Assert::IsTrue(true, message.get());
+            OutputDebugStringW(message.get());
```

Comment:
This is to aid debugging and verification. Will be rewritten when we address [Rewrite DynamicDependencies tests using TAEF #559](https://github.com/microsoft/ProjectReunion/issues/559)